storage-proofs/src/compound_proof.rs

```diff
@@ -268,12 +268,18 @@ pub trait CompoundProof<
 
     fn blank_circuit(public_params: &S::PublicParams) -> C;
 
-    fn groth_params(public_params: &S::PublicParams) -> Result<groth16::MappedParameters<E>> {
-        Self::get_groth_params(Self::blank_circuit(public_params), public_params)
+    fn groth_params<R: RngCore>(
+        rng: Option<&mut R>,
```

Comment:
lets add a doc comment clearly stating what happens if you pass an rng, vs not, to all these methods